thrift/types.go

```diff
@@ -115,11 +116,15 @@ func parseBinaryList(vl []interface{}) ([]byte, error) {
 	bs := make([]byte, 0, len(vl))
 	for _, v := range vl {
 		switch v := v.(type) {
+		// YAML uses int for all small values. For large values, it may use int64 or uint64
+		// but those are not valid bool values anyway, so we don't need to check them.
 		case int:
-			// YAML uses int for all small values. For large values, it may use int64 or uint64
-			// but those are not valid bool values anyway, so we don't need to check them.
-			if v < 0 || v >= (1<<8) {
-				return nil, fmt.Errorf("failed to parse list of bytes: %v is not a byte", v)
+			// from https://thrift.apache.org/docs/types, byte is an 8-bit signed integer
+			// For Go, thriftrw uses a Go byte, which is an uint8 so it is possible to have 0 to 255 over the wire.
+			// Apache Java clients respect the signed int8 and send [-128,127].
```

Comment:
let's remove "over the wire" since it's not that the numbers sent on the wire are different. over the wire, a byte is always the same -- it's just semantic meaning on the byte that languages put on top of the byte that results in negative values vs values > 128.
```suggestion
			// In Go, byte is an unsigned int8 (uint8) which has a range [0, 255].
			// In Java, byte is a signed int8 which has a range of [-128,127].
```